lib/archethic/contracts/interpreter/common_interpreter.ex

```diff
@@ -353,6 +360,80 @@ defmodule Archethic.Contracts.Interpreter.CommonInterpreter do
     {new_node, acc}
   end
 
+  # BigInt mathematics to avoid floating point issues
+  # the `0.0 + x` is used to cast integers to floats
+  def postwalk(_node = {:*, meta, [lhs, rhs]}, acc) do
+    new_node =
+      quote line: Keyword.fetch!(meta, :line) do
+        Decimal.to_float(
```

Comment:
You could create a private function to trunc the decimals as you use the same conversion on every function. Maybe somtheing like
```elixir
Decimal.mult(
  Decimal.from_float(0.0 + unquote(lhs)),
  Decimal.from_float(0.0 + unquote(rhs))
)
|> trunc_decimals()

# Maybe rename this function
defp trunc_decimals(val), do:
  val |> Decimal.round(8, :floor) |> Decimal.to_float()
```